doc/adr/2018-06-19-do-not-duplicate-server-side-validation.md

```diff
@@ -0,0 +1,24 @@
+**Title**: Do not duplicate validation performed by the server-side.
```

Comment:
Somewhere in each ADR should be the definition of "scope" of the ADR, e.g., does it apply to all C++ client libraries, or a specific product, or a set of products. This one sounds like it appears to all client libraries? The only other one is a GCS-only ADR.

Separately, I think the `doc/adr` directory should have a README file that's essentially the TOC for all of the files there, since it will be auto-displayed when viewing the directory, and is the most obvious place to go to look for the overview for the directory.